sample/07-geopip-blocking/src/geoip.interceptor.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { GeoIPInterceptor } from './geoip.interceptor';
+import { ExecutionContext, HttpException } from '@nestjs/common';
+import { Observable, of } from 'rxjs';
+
+describe('GeoIpInterceptor', () => {
+  let interceptor: GeoIPInterceptor;
+
+  beforeEach(() => {
+    interceptor = new GeoIPInterceptor(['India']);
+  });
+
+  it('should allow a request from India', async () => {
+    const executionContext = createMockExecutionContext('115.240.90.163'); // Mock IP from India
+    const callHandler = {
+      handle: jest.fn().mockReturnValue(of({})),
+    };
+
+    await expect(
+      interceptor.intercept(executionContext, callHandler),
+    ).resolves.toBeInstanceOf(Observable);
+  });
+
+  it('should allow a request from India with IPv6', async () => {
+    const executionContext = createMockExecutionContext(
+      '2401:4900:1c70:140b:fd0c:65a1:64e2:81bb',
+    ); // Mock IP from India
+    const callHandler = {
+      handle: jest.fn().mockReturnValue(of({})),
+    };
+
+    await expect(
+      interceptor.intercept(executionContext, callHandler),
+    ).resolves.toBeInstanceOf(Observable);
+  });
+
+  it('should block a request from outside India', async () => {
```

Comment:
add more test cases:

1. should block iPv6 outside india.
2. should allow IPv4 outside India.
3. should allow iPv6 outside India.